dev/com.ibm.ws.security.mp.jwt_fat/fat/src/com/ibm/ws/security/mp/jwt/fat/MPJwtConfigUsingBuilderTests.java

```diff
@@ -0,0 +1,1087 @@
+
+    static List<List<String>> testApps = null;
+
+    /**
+     * Startup the builder and resource servers
+     * Set flag to tell the code that runs between tests NOT to restore the server config between tests
+     * (very few of the tests use the config that the server starts with - this setting will save run time)
+     * Build the list of app urls and class names that each test will use
+     * 
+     * @throws Exception
+     */
+    @BeforeClass
+    public static void setUp() throws Exception {
+
+        setUpAndStartBuilderServer(jwtBuilderServer, "server_using_buildApp.xml", false);
+
+        setUpAndStartRSServerForTests(resourceServer, "rs_server_orig_withAudience.xml", false);
+
+        initTestAppArray();
+
+    }
+
+    /**
+     * Don't restore between tests
+     * Almost all of the tests in this class will reconfigure the server (we're testing mpJwt config and
+     * only 1 may exist in a server at a time, so, we need to reconfigure for each test)
+     * A few tests do use the configuration that the server starts with - they're do a normal config restore.
+     * All other will skip the restore by overriding the restoreTestServers method
+     */
+    @Override
+    public void restoreTestServers() {
+        Log.info(thisClass, "restoreTestServersWithCheck", "* Skipping server restore **");
+        logTestCaseInServerLogs("** Skipping server restore **");
+    }
+
+    /**
+     * Gets the resource server up and running.
+     * Sets properties in bootstrap.properties that will affect server behavior
+     * Sets up and installs the test apps
+     * Adds the server to the serverTracker (used for server restore and test class shutdown)
+     * Starts the server using the provided configuration file
+     * Saves the port info for this server (allows tests with multiple servers to know what ports each server uses)
+     * Allow some failure messages that occur during startup (they're ok and doing this prevents the test framework from failing)
+     * 
+     * @param server
+     *            - the server to process
+     * @param configFile
+     *            - the config file to start the server with
+     * @param jwkEnabled
+     *            - do we want jwk enabled (sets properties in bootstrap.properties that the configs will use)
+     * @throws Exception
+     */
+    protected static void setUpAndStartRSServerForTests(LibertyServer server, String configFile, boolean jwkEnabled) throws Exception {
+        bootstrapUtils.writeBootstrapProperty(server, MpJwtFatConstants.BOOTSTRAP_PROP_FAT_SERVER_HOSTNAME, getServerHostName());
+        bootstrapUtils.writeBootstrapProperty(server, MpJwtFatConstants.BOOTSTRAP_PROP_FAT_SERVER_HOSTIP, getServerHostIp());
+        if (jwkEnabled) {
+            bootstrapUtils.writeBootstrapProperty(server, "mpJwt_keyName", "");
+            bootstrapUtils.writeBootstrapProperty(server, "mpJwt_jwksUri", getServerSecureUrlBase(jwtBuilderServer) + "jwt/ibm/api/defaultJWT/jwk\"");
```

Comment:
Is `"jwt/ibm/api/defaultJWT/jwk\""` meant to have that escaped `"` character at the end?